File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    let ob = ['zero','one','two','three','four','five','six','seven','eight','nine']
    let answer = '';
    let temp = '';
    for(let i=0; i<s.length; i++){
        if(!isNaN(s[i])){
          answer+=s[i];   
        }else{
            temp+=s[i];
        }
        if(ob.indexOf(temp)!== -1 ){
            answer+=ob.indexOf(temp);
            temp = ''
        }
    }
    return +answer
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
import java.util.*;

class Solution {
    public int solution(String s) {
        List<String> list = 
            new ArrayList<>(List.of("zero","one","two","three","four","five","six","seven","eight","nine"));

        String answer = "";
        String temp = "";

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (!Character.isDigit(c)) {
                temp += c;
            } else {
                answer += c;
            }

            int index = list.indexOf(temp);
            if (index != -1) {
                answer += index;
                temp = "";
            }
        }
        
        return Integer.parseInt(answer);
    }
}